homeassistant/components/xbox/sensor.py

```diff
@@ -359,3 +441,58 @@ def native_value(self) -> StateType:
         """Return the state of the requested attribute."""
 
         return self.entity_description.value_fn(self.data) if self.data else None
+
+
+class XboxTitleHistorySensorEntity(
+    CoordinatorEntity[XboxTitleHistoryCoordinator], RestoreSensor
+):
+    """Title history sensor for the Xbox integration."""
+
+    _attr_has_entity_name = True
+    entity_description: XboxTitleHistorySensorEntityDescription
+    _restored_state: StateType = None
+    _restored_attributes: dict[str, Any] | None = None
+
+    def __init__(
+        self,
+        coordinator: XboxTitleHistoryCoordinator,
+        config_entry: XboxConfigEntry,
+        entity_description: XboxTitleHistorySensorEntityDescription,
+    ) -> None:
+        """Initialize the Xbox title history sensor."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._attr_unique_id = f"{config_entry.unique_id}_{entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, str(config_entry.unique_id))},
+            name=config_entry.title,
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Restore last state when entity is added."""
+        await super().async_added_to_hass()
+        if last_state := await self.async_get_last_state():
+            try:
+                self._restored_state = int(last_state.state)
+            except (ValueError, TypeError):
+                self._restored_state = None
+            self._restored_attributes = dict(last_state.attributes)
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        if self.coordinator.data is not None:
+            return self.entity_description.value_fn(self.coordinator.data)
+        # Use restored state if coordinator hasn't updated yet
+        return self._restored_state
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return additional attributes."""
+        if self.entity_description.attributes_fn:
+            # Prefer live coordinator data
+            if self.coordinator.data is not None:
+                return self.entity_description.attributes_fn(self.coordinator.data)
+            # Fall back to restored attributes until first update
+            return self._restored_attributes
+        return None
```

Comment:
Can we exclude the extra state attributes? Ideally we don't introduce big objects in them, or things that can be contextualized. Ideally it becomes a service, just like we have in overseerr, then users can make their own sensors using template triggers